icu4c/source/common/locid.cpp

```diff
@@ -778,6 +994,24 @@ Locale::minimizeSubtags(UErrorCode& status) {
     }
 }
 
+void
+Locale::canonicalize(UErrorCode& status) {
+    if (isBogus()) {
+        status = U_ILLEGAL_ARGUMENT_ERROR;
+    }
+    if (U_FAILURE(status)) {
+        return;
+    }
+    CharString uncanonicalize(fullName, status);
```

Comment:
Please add a 'd' to the end of this variable name.

Reply:
done